lib/normalizer.c

```diff
@@ -3452,12 +3452,15 @@ grn_nfkc_normalize_unify_iteration_mark(grn_ctx *ctx,
   *n_used_bytes = current_length;
   *n_used_characters = 1;
 
-  /* U+309D HIRAGANA ITERATION MARK */
-  if (previous_length > 0 && current_length == 3 && current[0] == 0xe3 &&
-      current[1] == 0x82 && current[2] == 0x9d) {
+  if (previous_length > 0) {
     const unsigned char *previous = current - previous_length;
```

Comment:
We can always define this if we don't touch it when `previous_length == 0`.

```suggestion
  const unsigned char *previous = current - previous_length;
```

Reply:
fix: [0f71092](https://github.com/groonga/groonga/pull/2513/commits/0f710925c112b5f3aeaf4eb01fef9cd14f2857f9) When the previous_length isn't over 0, we will skip the following condition. So we don't have to check it here.